### PR TITLE
feat: Add vis colors as ts constants

### PR DESCRIPTION
**Pull request by dvdanielamoitzi**
Added
- categorical vis colors
- sequential vis colors
- diverging vis colors
as typescript constants and exported them.

Although we already have the `getCssValue` function to retrieve those colors from css variables, I think those typescript variables make it easier to access it.

I also changed some existing, hard-coded code to use the new variables.

I have additionally extended the mantine types to support our custom colors added.
**To have the extended types support in our apps as well**, we need to add the following entry to the `include` array in the `tsconfig.json` file in all apps: `"node_modules/visyn_core/src/types.ts"`

### CODE REVIEW

File: src/utils/colors.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  '#fac7a9',
+  '#f99761',
+  '#e06d3b',
+  '#c2451a',
+  '#99230d',
+  '#6f0000',
+];
+
+// diverging color map red-turquois
+export const divergingRedTurquois = [
```
Comment (dvdanielamoitzi) [inline src/utils/colors.ts:34]:
```suggestion
export const divergingRedTurquoise = [
```

File: src/utils/colors.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  '#fac7a9',
+  '#f99761',
+  '#e06d3b',
+  '#c2451a',
+  '#99230d',
+  '#6f0000',
+];
+
+// diverging color map red-turquois
+export const divergingRedTurquoise = [
```
Comment (dvdanielamoitzi) [inline src/utils/colors.ts:34]:
```suggestion
export const divergingRedTurquois = [
```

File: src/@types/mantine-overrides.d.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { DefaultMantineColor, MantineColorsTuple } from '@mantine/core';
+
+type ExtendedCustomColors = 'dvPrimary' | 'dvGray' | 'dvGene' | 'dvDisease' | 'dvCellLine' | 'dvTissue' | 'dvDrug' | 'white' | DefaultMantineColor;
+
+declare module '@mantine/core' {
```
Comment (puehringer) [inline src/@types/mantine-overrides.d.ts:5]: Do you know if this works transitively, i.e. for packages using `visyn_core`? If yes, that would be fantastic, as we wouldn't have to redefine the types everywhere.

Comment (dvdanielamoitzi) [reply src/@types/mantine-overrides.d.ts:5]: @puehringer I just tried that in BioInsight but it seems like this does not work transitively unfortunately, but I'll take a look and find out if we can do that somehow.

File: src/types.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import type { DefaultMantineColor, MantineColorsTuple } from '@mantine/core';
+
+type ExtendedCustomColors = 'dvPrimary' | 'dvGray' | 'dvGene' | 'dvDisease' | 'dvCellLine' | 'dvTissue' | 'dvDrug' | 'white' | DefaultMantineColor;
```
Comment (puehringer) [inline src/types.ts:3]: @dvdanielamoitzi `dvAI` is missing here.

File: tests/exports.test.ts [hunk 5]
```diff
@@ -4,8 +4,8 @@
 import { readdirSync, readFileSync } from 'fs';
 import packageJson from '../package.json';
 
-const NOT_EXPORTED_PACKAGES = ['assets', 'demo', 'locales', 'scss', 'stories'];
-const ADDITIONAL_EXPORTS = ['.', './assets/*', './plotly/full', './scss/*', './phovea_registry', './package.json'];
+const NOT_EXPORTED_PACKAGES = ['assets', 'demo', 'locales', 'scss', 'stories', '@types'];
```
Comment (puehringer) [inline tests/exports.test.ts:7]: @dvdanielamoitzi  `@types` doesn't exist anymore.

**Review by puehringer — APPROVED** [review]

**Pull request merged**